service/src/test/java/tds/assessment/repositories/impl/AssessmentMapperTest.java

```diff
@@ -247,6 +261,44 @@ public void shouldReturnASegmentedAdaptiveAssessment() {
         // all other fields are identical to firstSegment
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionForNullFtStartPositions() {
+        String assessmentKey = "nullFTStartPosAssessment";
+        List<Map<String, Object>> records =
+            AssessmentRecordBuilder.createSegmentedAssessmentWithManySelectionAlgorithms(assessmentKey,
+                1,
+                3,
+                fieldTestStartDate,
+                fieldTestEndDate,
+                segmentFieldTestStartDate,
+                segmentFieldTestEndDate,
+                null,
+                null);
+
+        Optional<Assessment> maybeAssessment = mapper.mapResults(records);
+
+        assertThat(maybeAssessment).isPresent();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionForZeroFtStartPositions() {
+        String assessmentKey = "nullFTStartPosAssessment";
+        List<Map<String, Object>> records =
+            AssessmentRecordBuilder.createSegmentedAssessmentWithManySelectionAlgorithms(assessmentKey,
+                1,
+                3,
+                fieldTestStartDate,
+                fieldTestEndDate,
+                segmentFieldTestStartDate,
+                segmentFieldTestEndDate,
+                0,
+                0);
+
+        Optional<Assessment> maybeAssessment = mapper.mapResults(records);
+
+        assertThat(maybeAssessment).isPresent();
```

Comment:
These `assertThat()` calls aren't needed right as it will throw the exception before it gets here.  Same for the previous test.